Regression: glfw build on Mac is failing to build

I just want a place to track this. If you remove all caches on macOS and build GLFW since https://github.com/ghostty-org/ghostty/commit/221f905a1c5332a96f04918ea87d1c85a05af035 you get errors like the following:

```
zig build -Dapp-runtime=glfw
```

```
install
└─ install ghostty
   └─ zig build-exe ghostty Debug aarch64-macos.13.0 2 errors
pkg/macos/foundation/c.zig:1:15: error: C import failed: FileNotFound
pub const c = @cImport({
              ^~~~~~~~
pkg/macos/text/c.zig:1:15: error: C import failed: FileNotFound
pub const c = @cImport({
              ^~~~~~~~
error: the following command failed with 2 compilation errors:
```

I have no idea why this is happening yet.

Note this applies _only to GLFW builds_. Standard libghostty and Swift builds (plain `zig build`) work fine.

Introduced by https://github.com/ghostty-org/ghostty/commit/221f905a1c5332a96f04918ea87d1c85a05af035#diff-201b1fb9d546eb76c303bb5e2c3bc04204496e3e4e2d8c68dbd7b7ecab440a03

--- a/pkg/macos/animation/c.zig
+++ b/pkg/macos/animation/c.zig
@@ -1,3 +1,1 @@
-pub const c = @cImport({
-    @cInclude("QuartzCore/CALayer.h");
-});
+pub const c = @import("../main.zig").c;

--- a/pkg/macos/build.zig
+++ b/pkg/macos/build.zig
@@ -45,10 +45,8 @@ pub fn build(b: *std.Build) !void {
         module.linkFramework("CoreVideo", .{});
         module.linkFramework("QuartzCore", .{});
 
-        if (!target.query.isNative()) {
-            try apple_sdk.addPaths(b, lib.root_module);
-            try apple_sdk.addPaths(b, module);
-        }
+        try apple_sdk.addPaths(b, lib.root_module);
+        try apple_sdk.addPaths(b, module);
     }
     b.installArtifact(lib);
 
@@ -59,9 +57,20 @@ pub fn build(b: *std.Build) !void {
             .target = target,
             .optimize = optimize,
         });
+        if (target.result.os.tag.isDarwin()) {
+            try apple_sdk.addPaths(b, test_exe.root_module);
+        }
         test_exe.linkLibrary(lib);
+
         var it = module.import_table.iterator();
-        while (it.next()) |entry| test_exe.root_module.addImport(entry.key_ptr.*, entry.value_ptr.*);
+        while (it.next()) |entry| {
+            test_exe.root_module.addImport(
+                entry.key_ptr.*,
+                entry.value_ptr.*,
+            );
+        }
+
+        b.installArtifact(test_exe);
 
         const tests_run = b.addRunArtifact(test_exe);
         const test_step = b.step("test", "Run tests");

--- a/pkg/macos/build.zig.zon
+++ b/pkg/macos/build.zig.zon
@@ -1,6 +1,7 @@
 .{
-    .name = "macos",
+    .name = .macos,
     .version = "0.1.0",
+    .fingerprint = 0x45e2f6107d5b2b2c,
     .paths = .{""},
     .dependencies = .{
         .apple_sdk = .{ .path = "../apple-sdk" },

--- a/pkg/macos/carbon/c.zig
+++ b/pkg/macos/carbon/c.zig
@@ -1,3 +1,1 @@
-pub const c = @cImport({
-    @cInclude("Carbon/Carbon.h");
-});
+pub const c = @import("../main.zig").c;

--- a/pkg/macos/dispatch/c.zig
+++ b/pkg/macos/dispatch/c.zig
@@ -1,3 +1,1 @@
-pub const c = @cImport({
-    @cInclude("dispatch/dispatch.h");
-});
+pub const c = @import("../main.zig").c;

--- a/pkg/macos/foundation/attributed_string.zig
+++ b/pkg/macos/foundation/attributed_string.zig
@@ -67,7 +67,7 @@ pub const MutableAttributedString = opaque {
     ) void {
         const T = @TypeOf(key);
         const info = @typeInfo(T);
-        const Key = if (info != .Pointer) T else info.Pointer.child;
+        const Key = if (info != .pointer) T else info.pointer.child;
         const key_arg = if (@hasDecl(Key, "key"))
             key.key()
         else

--- a/pkg/macos/foundation/c.zig
+++ b/pkg/macos/foundation/c.zig
@@ -1,3 +1,1 @@
-pub const c = @cImport({
-    @cInclude("CoreFoundation/CoreFoundation.h");
-});
+pub const c = @import("../main.zig").c;

--- a/pkg/macos/graphics/bitmap_context.zig
+++ b/pkg/macos/graphics/bitmap_context.zig
@@ -38,13 +38,13 @@ test {
     const cs = try graphics.ColorSpace.createDeviceGray();
     defer cs.release();
     const ctx = try BitmapContext.create(null, 80, 80, 8, 80, cs, 0);
-    defer ctx.release();
-
-    ctx.setShouldAntialias(true);
-    ctx.setShouldSmoothFonts(false);
-    ctx.setGrayFillColor(1, 1);
-    ctx.setGrayStrokeColor(1, 1);
-    ctx.setTextDrawingMode(.fill);
-    ctx.setTextMatrix(graphics.AffineTransform.identity());
-    ctx.setTextPosition(0, 0);
+    const context = BitmapContext.context;
+    defer context.release(ctx);
+    context.setShouldAntialias(ctx, true);
+    context.setShouldSmoothFonts(ctx, false);
+    context.setGrayFillColor(ctx, 1, 1);
+    context.setGrayStrokeColor(ctx, 1, 1);
+    context.setTextDrawingMode(ctx, .fill);
+    context.setTextMatrix(ctx, graphics.AffineTransform.identity());
+    context.setTextPosition(ctx, 0, 0);
 }

--- a/pkg/macos/graphics/c.zig
+++ b/pkg/macos/graphics/c.zig
@@ -1,3 +1,1 @@
-pub const c = @cImport({
-    @cInclude("CoreGraphics/CoreGraphics.h");
-});
+pub const c = @import("../main.zig").c;

--- a/pkg/macos/graphics/path.zig
+++ b/pkg/macos/graphics/path.zig
@@ -3,7 +3,7 @@ const assert = std.debug.assert;
 const Allocator = std.mem.Allocator;
 const foundation = @import("../foundation.zig");
 const graphics = @import("../graphics.zig");
-const c = @import("c.zig");
+const c = @import("c.zig").c;
 
 pub const Path = opaque {
     pub fn createWithRect(

--- a/pkg/macos/main.zig
+++ b/pkg/macos/main.zig
@@ -1,3 +1,5 @@
+const builtin = @import("builtin");
+
 pub const carbon = @import("carbon.zig");
 pub const foundation = @import("foundation.zig");
 pub const animation = @import("animation.zig");
@@ -6,6 +8,23 @@ pub const graphics = @import("graphics.zig");
 pub const os = @import("os.zig");
 pub const text = @import("text.zig");
 pub const video = @import("video.zig");
+
+// All of our C imports consolidated into one place. We used to
+// import them one by one in each package but Zig 0.14 has some
+// kind of issue with that I wasn't able to minimize.
+pub const c = @cImport({
+    @cInclude("CoreFoundation/CoreFoundation.h");
+    @cInclude("CoreGraphics/CoreGraphics.h");
+    @cInclude("CoreText/CoreText.h");
+    @cInclude("CoreVideo/CoreVideo.h");
+    @cInclude("QuartzCore/CALayer.h");
+    @cInclude("dispatch/dispatch.h");
+    @cInclude("os/log.h");
+
+    if (builtin.os.tag == .macos) {
+        @cInclude("Carbon/Carbon.h");
+    }
+});
 
 test {
     @import("std").testing.refAllDecls(@This());

--- a/pkg/macos/os/c.zig
+++ b/pkg/macos/os/c.zig
@@ -1,3 +1,1 @@
-pub const c = @cImport({
-    @cInclude("os/log.h");
-});
+pub const c = @import("../main.zig").c;

--- a/pkg/macos/text/c.zig
+++ b/pkg/macos/text/c.zig
@@ -1,3 +1,1 @@
-pub const c = @cImport({
-    @cInclude("CoreText/CoreText.h");
-});
+pub const c = @import("../main.zig").c;

--- a/pkg/macos/text/font.zig
+++ b/pkg/macos/text/font.zig
@@ -280,7 +280,8 @@ test {
         const cs = try graphics.ColorSpace.createDeviceGray();
         defer cs.release();
         const ctx = try graphics.BitmapContext.create(null, 80, 80, 8, 80, cs, 0);
-        defer ctx.release();
+        const context = graphics.BitmapContext.context;
+        defer context.release(ctx);
 
         var pos = [_]graphics.Point{.{ .x = 0, .y = 0 }};
         font.drawGlyphs(

--- a/pkg/macos/text/font_descriptor.zig
+++ b/pkg/macos/text/font_descriptor.zig
@@ -276,7 +276,7 @@ test "descriptor" {
     const v = try FontDescriptor.createWithNameAndSize(name, 12);
     defer v.release();
 
-    const copy_name = v.copyAttribute(.name);
+    const copy_name = v.copyAttribute(.name).?;
     defer copy_name.release();
 
     {

--- a/pkg/macos/video/c.zig
+++ b/pkg/macos/video/c.zig
@@ -1,3 +1,1 @@
-pub const c = @cImport({
-    @cInclude("CoreVideo/CoreVideo.h");
-});
+pub const c = @import("../main.zig").c;

